frontend/src/metabase/lib/formatting.js

```diff
@@ -250,23 +250,35 @@ function formatNumberScientific(
   }
 }
 
+const DISPLAY_COMPACT_DECIMALS_CUTOFF = 1000;
+export const COMPACT_CURRENCY_OPTIONS = {
+  // Currencies vary in how many decimals they display, so this is probably
+  // wrong in some cases. Intl.NumberFormat has some of that data built-in, but
+  // I couldn't figure out how to use it here.
+  digits: 2,
+  currency_style: "symbol",
+  // We need this to ensure the settings are used. Otherwise, a cached
+  // _numberFormatter would take precedence.
+  _numberFormatter: undefined,
```

Comment:
Hmm won't this always disable the `_numberFormatter` optimization for currency?

Actually, maybe a better approach to that optimization would be to just cache `numberFormatterForOptions` with the attributes we pass to `new Intl.NumberFormat` as the key,  but we should probably benchmark it to make sure it's fast enough.

Reply:
It disables it for compact currency, which should be fine since we only use compact formatting in low-cardinality places: scalars on dashboards, choropleths, and chart values.

Though, I'm realizing that `_numberFormatter: undefined` should really only be used where we test compact/non-compact lengths. It's not a `COMPACT_CURRENCY_OPTIONS` but rather something specific to testing different options. I just moved it there.

I like your idea for improving how we do this optimization though. I'll make an issue for that as a small refactor.